### ISSUE TITLE
setting `RotateLogOnStartup: true` attribute doesnot have any effect

### ISSUE THREAD

**Issue by zeusbaba** [oracle/weblogic-deploy-tooling#879]
When configuring OSB domain, I would like to customise `Log:` attribute with extended params.  
One of those params doesnot have any effect, aka it doesnot set the server's logging config.  

See   
```
Log:
        FileCount: 20
        LogFileRotationDir: /u06/{{ oracle_user }}/logs/rotatedLogs
        RotateLogOnStartup: true
        RotationType: bySize
        NumberOfFilesLimited: true
        RedirectStderrToServerLogEnabled: true
        RedirectStdoutToServerLogEnabled: true
        FileMinSize: 5000
        DomainLogBroadcasterBufferSize: 1
        FileName: {{ domain_logs_dir }}.log
```   
When check AdminServer console, I see that all attributes are actively set, except `RotateLogOnStartup: true`  which does no effect on the server's Log configuration, aka `Rotate log file on startup` is still inactive.    

- WDT version: `20210402 built from master branch`  
- OSB version: `12.2.1.4`  

Can you please advise/help?

**Comment by CarolynRountree** [comment]
RotateLogOnStartup default value is true. Therefore you might not see it in the config.xml, but the value is inactive. Can you advise that the log is not rotated on startup? In which case this might be a WLST bug.

**Comment by zeusbaba** [comment]
thnx for your quick response.  
In our installations `RotateLogOnStartup` default value seems to be `false` , because when we don't set this param explicitly, I can see that AdminConsole still displays it as `inactive`.   
Using WDT we want to set it to active by using `RotateLogOnStartup: true` attribute, but no avail.  

I'll few more tests today; restarting and checking logs, and afterwards I'll share more details.

**Comment by zeusbaba** [comment]
I just tested by restarting ManagedServer and observed that logs ARE rotated.  
Strangely, AdminServer console still shows this option as `inactive` , see screenshot of the server console which still displays `Rotate log file on startup` option as `inactive` even though I did explicitly set this attribute via WDT.   

![Skjermbilde 2021-04-28 092648](https://user-images.githubusercontent.com/397125/116364575-9f3f6680-a804-11eb-8430-2c3f75292996.png)

**Comment by zeusbaba** [comment]
while checking WDT codebase I stumbled upon [this line](https://github.com/oracle/weblogic-deploy-tooling/blob/baf0b2ca9c9ebdd68f2cc0c35e9e823e505b51bc/core/src/main/resources/oracle/weblogic/deploy/aliases/category_modules/ServerTemplate.json#L274)   
does this hint that `RotateLogOnStartup` attribute is support _only_ for `"version": "[12.1.2,)",`  ?

**Comment by zeusbaba** [comment]
I've just tried [this workaround](https://github.com/oracle/weblogic-deploy-tooling/issues/304#issuecomment-476768295) which was mentioned in #304 
  
The result is;  
`update-domain-online` does correctly set `RotateLogOnStartup: true` **only** for `topology:/Log` attributes and it completely skips the ones defined under `topology:/Server/AdminServer/Log` or `topology:/Server/ms_server/Log`

**Comment by CarolynRountree** [comment]
I do not see where [12.1.2,) is set for Server/Log/RotateLogOnStartup. Here is what is set, and true is also the default for Server rotateLogOnStartup
```
                "RotateLogOnStartup":                            [ {"version": "[10,)",                "wlst_mode": "both",    "wlst_name": "RotateLogOnStartup",                            "wlst_path": "WP001", "value": {"default": "true"  }, "wlst_type": "boolean"} ],
```
However, [12.1.2,) means that the atttibute first became effective in 12.1.2 and is in all further versions

**Comment by zeusbaba** [comment]
> I do not see where [12.1.2,) is set for Server/Log/RotateLogOnStartup. Here is what is set, and true is also the default for Server rotateLogOnStartup
> 
> ```
>                 "RotateLogOnStartup":                            [ {"version": "[10,)",                "wlst_mode": "both",    "wlst_name": "RotateLogOnStartup",                            "wlst_path": "WP001", "value": {"default": "true"  }, "wlst_type": "boolean"} ],
> ```
> 
> However, [12.1.2,) means that the atttibute first became effective in 12.1.2 and is in all further versions

thanks for clarification

**Comment by zeusbaba** [comment]
> I've just tried [this workaround](https://github.com/oracle/weblogic-deploy-tooling/issues/304#issuecomment-476768295) which was mentioned in #304
> 
> The result is;
> `update-domain-online` does correctly set `RotateLogOnStartup: true` **only** for `topology:/Log` attributes and it completely skips the ones defined under `topology:/Server/AdminServer/Log` or `topology:/Server/ms_server/Log`

In addition to that, I've manually tried using WLST-online (without WDT) and successfully did set  
```
   startEdit()  
Started edit session ...  
  
  cd("Servers/AdminServer/Log/AdminServer")  
wls:/theosb_domain/edit/Servers/AdminServer/Log/AdminServer !>  
  cmo.setRotateLogOnStartup(true)  

  save()  
Saving all your changes ...  
Saved all your changes successfully.
  activate()
```  

I _suspect_  that using WDT to updateDomain (online) might have issues. see my previous message.  
Can you please advise?

**Comment by CarolynRountree** [comment]
I was not able to recreate the problem online with the Server/Log. As you can see in the following snippet, the value is explicitly set.

As per the mentioned issue, do really fix this problem, you will need to open a problem with WebLogic Server for WLST.

```  <server>
    <name>AdminServer</name>
    <log>
      <date-format-pattern>MMM d, yyyy h:mm:ss,SSS a z</date-format-pattern>
      <file-name>logs/AdminServer.log</file-name>
      <rotate-log-on-startup>true</rotate-log-on-startup>

**Comment by ddsharpe** [comment]
@zeusbaba This appears to be a duplicate of #938.  As I posted there, https://github.com/oracle/weblogic-deploy-tooling/issues/938#issuecomment-896281681
This should be fixed in the current release, and I can no longer reproduce it.

**Comment by ddsharpe** [comment]
@zeusbaba This is a WLST offline bug in WebLogic.  The bug that I filed on your behalf is fixed in the next production release of WebLogic Server.  To get this fix for the release of WLS that you are using, you will need to file a support case with Oracle technical support and ask that the fix be ported to your specific release.  Be sure to mention, in your support case, all one-off patches and PSUs that you are using. The bug number of the original fix is 29547985.
